src/test/passes/strings-encoding/basic-aarch64-android.cpp

```diff
@@ -0,0 +1,29 @@
+// REQUIRES: aarch64-registered-target
+
+// The default object contains the file-name string:
+//     RUN: clang++ -target aarch64-linux-android -O1 -c %s -o - | strings | FileCheck --check-prefix=CHECK-DEFAULT -DFILE_NAME=%s %s
+//     CHECK-DEFAULT: [[FILE_NAME]]
+
+// The 'remove' configuration overwrites it with the fixed REDACTED literal:
+//     RUN: env OMVLL_CONFIG=%S/config_remove.py clang++ -target aarch64-linux-android -fpass-plugin=%libOMVLL -O1 -c %s \
+//     RUN:                                              -o - | strings | FileCheck --check-prefix=CHECK-REMOVED -DFILE_NAME=%s %s
+//     CHECK-REMOVED-NOT: [[FILE_NAME]]
+//     CHECK-REMOVED:     REDACTED
+
+// The 'replace' configuration encodes the string and adds logic that decodes it at load-time:
+//     RUN: env OMVLL_CONFIG=%S/config_replace.py clang++ -target aarch64-linux-android -fpass-plugin=%libOMVLL -O1 -c %s \
+//     RUN:                                               -o - | strings | FileCheck --check-prefix=CHECK-REPLACED -DFILE_NAME=%s %s
```

Comment:
This crashes on macOS with AppleClang 1400.0.29.202 (Xcode 14.1 and 14.2). I didn't investigate yet, but maybe related to your comment in [src/core/Jitter.cpp#L295](https://github.com/open-obfuscator/o-mvll/blob/1.0.1/src/core/Jitter.cpp#L295) @romainthomas ?

The diagnostic `Illegal instruction: 4` isn't very helpful, but let me still attach the crash log to clarify the repro: 
[string-encoding-replace-jit-crash.log](https://github.com/open-obfuscator/o-mvll/files/11564275/string-encoding-replace-jit-crash.log)

Reply:
Yes there are definitely some issues with the JIT on OSX but I didn't investigated the root cause.